language/compiler/ir_to_bytecode/src/compiler.rs

```diff
@@ -254,9 +253,14 @@ type FieldMap = HashMap<String, FieldDefinitionIndex>;
 struct CompilationScope<'a> {
     // maps from handles in the compilation unit (reference) to the definitions
     // in the imported unit (definitions).
+    //
+    // TODO: module handle indices should not appear here as it only makes sense within the context
+    //       of the module currently being compiled (leaky abstraction)
     imported_modules: HashMap<String, (ModuleIndex, ModuleHandleIndex)>,
     function_definitions:
         HashMap<ModuleHandleIndex, HashMap<String, (ModuleIndex, FunctionDefinitionIndex)>>,
+    struct_definitions: HashMap<String, HashMap<String, StructDefinitionIndex>>,
```

Comment:
Should we take a `ModuleHandleIndex` instead of a `String`? This would keep it consistent with `function_definitions`

Reply:
That's a good point. However as the comment I added says, I think it is a bad idea to have ModuleHandleIndex here at all. Let me explain: the ModuleHandleIndex is local to the module currently being compiled, which is out of the this scope. (A compilation scope contains only information about the bigger context, i.e. the dependent modules.) 

Taking a ModuleHandleIndex actually creates unnecessary problems for ourselves. If we look at function_definitions we'll see that it has to store a ModuleIndex in the value. This is because a ModuleHandleIndex acts just like a black box, giving no useful information locating the module we want in the current scope.

From my opinion, it would be better we do it the other way around: change ModuleHandleIndex to String in function_definitions.

Reply:
What about `ModuleId` (which is fully qualified) instead of `String` (or at least some type aliases to document what the string is)?

Reply:
That sounds reasonable. Let me make the changes.